var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -227,6 +235,25 @@ def configure_args(self):
         extra_args += self.enable_or_disable('hl')
         extra_args += self.enable_or_disable('fortran')
 
+        # Set the api version
+        if 'api=none' in self.spec:
+            extra_args.append('')
+
+        if 'api=v114' in self.spec:
+            extra_args.append('--with-default-api-version=v114')
+
+        if 'api=v112' in self.spec:
+            extra_args.append('--with-default-api-version=v112')
+
+        if 'api=v110' in self.spec:
+            extra_args.append('--with-default-api-version=v110')
+
+        if 'api=v18' in self.spec:
+            extra_args.append('--with-default-api-version=v18')
+
+        if 'api=v16' in self.spec:
+            extra_args.append('--with-default-api-version=v16')
```

Comment:
I simpler way to do this would be:
```python
api = self.spec.variants['api'].value
if api != 'none':
    extra_args.append('--with-default-api-version=' + api)
```